src/main/java/at/hannibal2/skyhanni/data/model/TabWidget.kt

```diff
@@ -371,10 +375,21 @@ enum class TabWidget(
         /** Patterns that where loaded from a future version*/
         private var extraPatterns: List<Pattern> = emptyList()
 
+        private var sentSinceWorldChange = false
+
         init {
             entries.forEach { it.pattern }
         }
 
+        @SubscribeEvent
+        fun onSecond(event: SecondPassedEvent) {
+            if (sentSinceWorldChange || !LorenzUtils.inSkyBlock) return
+            sentSinceWorldChange = true
+            @Suppress("DEPRECATION")
+            update(TabListData.getTabList())
+            ChatUtils.debug("Forcefully Updated Widgets")
+        }
+
```

Comment:
Here you gamble that the tablist is already correct. But this might not be the case at this time. Since full world switches can take longer than 1 second (with a bit of lag).